### PR TITLE
Add connection metrics per username

### PR DESCRIPTION
**Pull request by antoinebhs**
Use multigauge to handle connection metrics per username.  

From micrometer documentation, the multigauge is the best fit for this use:
`Micrometer supports one last special type of Gauge, called a MultiGauge, to help manage gauging a growing or shrinking list of criteria. `
Source: https://docs.micrometer.io/micrometer/reference/concepts/gauges.html

### CODE REVIEW

File: src/main/java/org/gridsuite/study/notification/server/NotificationWebSocketHandler.java [hunk 1]
```diff
@@ -235,11 +235,19 @@ private void updateConnectionMetrics(WebSocketSession webSocketSession) {
...
 
     private void updateDisconnectionMetrics(WebSocketSession webSocketSession) {
         var userId = webSocketSession.getHandshakeInfo().getHeaders().getFirst(HEADER_USER_ID);
         LOGGER.info("Websocket disconnection id={} for user={}", webSocketSession.getId(), userId);
         userConnections.computeIfPresent(userId, (k, v) -> v > 1 ? v - 1 : null);
+        updateConnectionMetricsRegistry();
+    }
+
+    private void updateConnectionMetricsRegistry() {
+        multiGauge.register(userConnections.entrySet().stream()
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/notification/server/NotificationWebSocketHandler.java:249]: Perhaps add here the builder ?
`MultiGauge.builder(USERS_METER_NAME).register(meterRegistry)`

File: src/test/java/org/gridsuite/study/notification/server/NotificationWebSocketIT.java [hunk 2]
```diff
@@ -50,27 +49,47 @@ public void echo() {
...
+        Mono.zip(connection1, connection2, connection3).block();
     }
 
-    private void testMeter(String name, int val) {
-        Gauge meter = meterRegistry.get(name).gauge();
-        assertNotNull(meter);
-        assertEquals(val, Double.valueOf(meter.value()).intValue());
+    private void testMeterMap(Map<String, Double> userMap) {
+        for (Map.Entry<String, Double> userEntry : userMap.entrySet()) {
+            assertEquals(userEntry.getValue(), meterRegistry.get(USERS_METER_NAME).tag(USER_TAG, userEntry.getKey()).gauge().value(), 1e-6);
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/notification/server/NotificationWebSocketIT.java:92]: `1e-6` ?

File: src/main/java/org/gridsuite/study/notification/server/NotificationWebSocketHandler.java [hunk 3]
```diff
@@ -75,23 +78,20 @@ public class NotificationWebSocketHandler implements WebSocketHandler {
     static final String HEADER_INDEXATION_STATUS = "indexation_status";
 
     static final String USERS_METER_NAME = "app.users";
-    static final String CONNECTIONS_METER_NAME = "app.connections";
+    static final String USER_TAG = "user";
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/notification/server/NotificationWebSocketHandler.java:81]: user**s** ?

Comment (antoinebhs) [reply src/main/java/org/gridsuite/study/notification/server/NotificationWebSocketHandler.java:81]: This identify each user, see the /actuator/prometheus endpoint so I think we should stick to "user":
```
# HELP app_users Number of connections per user
# TYPE app_users gauge
app_users{user="user1",} 1.0
app_users{user="user2",} 1.0
```
It's also more logical when used in grafana, we use `sum by (user) (app_users)`. Plural form does not make sense here.

**Pull request merged**